src/async_client.rs

```diff
@@ -227,46 +233,46 @@ impl Drop for AsyncClient {
     }
 }
 
-async fn background_task(client: Arc<Mutex<ua::Client>>, cycle_time: time::Duration) {
+async fn background_task(client: Arc<Mutex<ua::Client>>, cycle_time: Duration) {
     log::debug!("Starting background task");
 
     let mut interval = time::interval(cycle_time);
-    // TODO: Customized MissedTickBehavior? Only Skip and Delay are suitable here.
-    interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
-    let mut last_tick;
+    // TODO: Offer customized `MissedTickBehavior`? Only `Skip` and `Delay` are suitable here as we
+    // don't want `Burst` to repeatedly and unnecessarily call `UA_Client_run_iterate()` many times
+    // in a row.
+    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
 
-    // `UA_Client_run_iterate()` must be run periodically and makes sure to
-    // maintain the connection (e.g. renew session) and run callback handlers.
+    // `UA_Client_run_iterate()` must be run periodically and makes sure to maintain the connection
+    // (e.g. renew session) and run callback handlers.
     loop {
-        log::trace!("Next iterate cycle");
-        let last_cycle = time::Instant::now();
+        // This await point is where the background task could be aborted. (The first tick finishes
+        // immediately, so there is no additional delay on the first iteration.)
+        interval.tick().await;
```

Comment:
AFAIK `tick()` should return the time at which the tick should have occurred, i.e. the scheduled instant. But I couldn't find the corresponding reference in the docs again.